mmv1/provider/core.rb

```diff
@@ -190,12 +190,72 @@ def compile_file_list(output_folder, files, file_template, pwd = Dir.pwd)
           Google::LOGGER.debug "Compiling #{source} => #{target}"
           file_template.generate(pwd, source, target, self)
 
+          add_hashicorp_copyright_header(output_folder, target)
           replace_import_path(output_folder, target)
         end
       end.map(&:join)
       Dir.chdir pwd
     end
 
+    def add_hashicorp_copyright_header(output_folder, target)
+      # only add copyright headers when generating TPG and TPGB
+      return unless output_folder.end_with?('terraform-provider-google') ||
+                    output_folder.end_with?('terraform-provider-google-beta')
+
+      # Prevent adding copyright header to files with paths or names matching the strings below
+      # NOTE: these entries need to match the content of the .copywrite.hcl file originally
+      #       created in https://github.com/GoogleCloudPlatform/magic-modules/pull/7336
+      ignored_folders = [
+        '.github/',
+        '.release/',
+        '.changelog/',
+        'examples/',
+        'META.d/'
+      ]
+      ignored_files = [
+        'go.mod',
+        '.goreleaser.yml',
+        '.golangci.yml',
+        'terraform-registry-manifest.json'
+      ]
+      should_add_header = true
+      ignored_folders.each do |folder|
+        # folder will be path leading to file
+        next unless target.start_with? folder
+
+        Google::LOGGER.debug 'Not adding HashiCorp copyright headers in ' \
```

Comment:
INFO appears on every run but DEBUG won't, right?

Reply:
Yeah that's correct - viewing the debug logs requires passing in `-d`/`--debug` flags to the compiler and info is visible by default/without needing to edit the Makefile